pandas/io/stata.py

```diff
@@ -644,12 +649,22 @@ def __init__(self, catarray: Series, encoding: str = "latin-1"):
         self.labname = catarray.name
         self._encoding = encoding
         categories = catarray.cat.categories
-        self.value_labels = list(zip(np.arange(len(categories)), categories))
+        self.value_labels: list[tuple[int | float, str], ...] = list(
+            zip(np.arange(len(categories)), categories)
+        )
         self.value_labels.sort(key=lambda x: x[0])
+
         self.text_len = 0
         self.txt: list[bytes] = []
         self.n = 0
+        self.off = np.array([])
+        self.val = np.array([])
+        self.len = 0
 
+        self._prepare_value_labels()
+
+    def _prepare_value_labels(self):
```

Comment:
I had originally copied the whole init from `StataValueLabel` and only changed where the label name and values were coming from. I moved the encoding of the labels to a separate function here to avoid repeating that process and to make sure both are encoded in the same way
Raising the type error makes sense to me, but it would change the existing categorical encoding. I can remove the addition of `_prepare_value_labels` and only raise the error in the new StataNonCatValueLabel, or make the change in both?

Reply:
Hey @bashtage, I realized that I didn't respond directly to your comments. Does this make sense?

Reply:
Yes, it seems clear now. Thanks.